src/rpc/rpc.cc

```diff
@@ -643,6 +706,12 @@ namespace rpc {
           auto size = read_le<uint32_t>(ptr + 8);
           return std::make_tuple(msgid, size);
       }
+      static void encode_header(int64_t msg_id, snd_buf& data) {
+          static_assert(snd_buf::chunk_size >= raw_header_size, "send buffer chunk size is too small");
+          auto p = data.front().get_write();
+          write_le<int64_t>(p, msg_id);
+          write_le<uint32_t>(p + 8, data.size - raw_header_size);
```

Comment:
shall we replace 8 with `sizeof(int64_t)` here? or, can we update `write_le<>`, so it looks something like:
```c++
template <typename T>
inline
char*
write_le(char* p, T datum) noexcept {
    datum = cpu_to_le(datum);
    return std::copy_n(reinterpret_cast<const char*>(&datum), sizeof(T), p);
}
```

so we can have something like:
```c++
auto p = data.front().get_write();
p = write_le<int64_t>(p, msg_id);
p = write_le<uint32_t>(p, data.size - raw_header_size);
```

?